### ISSUE TITLE
Imported views not analysed

### ISSUE THREAD

**Issue by LeFrosch** [simolus3/moor#1639]
**Describe the bug**
When views are imported from another drift file, they are not analysed. Thus all column references will fail. For example:
File: test.drift
```sql
CREATE TABLE profile (
    uuid TEXT NOT NULL,
    ...
);

CREATE VIEW test_view AS SELECT * FROM profile;
```
File: test_view.drift
```sql
import 'test.drift';

useView2(REQUIRED :uuid):
SELECT * FROM test_view WHERE uuid = :uuid;
```
Returns the following error:
```
line 4, column 31: Unknown column. These columns are available: 
  ╷
4 │ SELECT * FROM test_view WHERE uuid = :uuid;
  │                               ^^^^
  ╵
```
I think this could be caused by not analysing imported views. Because in `drift_dev/lib/src/analyzer/runner/steps/analyze_moor.dart:37` the ViewAnalyser only resolves declared views but imported ones are not resolved:
```dart
await ViewAnalyzer(
        this, availableTables, importedViews, parseResult.imports)
    .resolve(parseResult.declaredViews);
```
The following change fixes the problem:
```dart
await ViewAnalyzer(
        this, availableTables, importedViews, parseResult.imports)
    .resolve([...parseResult.declaredViews, ...importedViews]);
```
But this will analyse some views twice and idk if this has any side effects.

**Comment by simolus3** [comment]
> But this will analyse some views twice and idk if this has any side effects.

If there are any lints/analysis errors in the view, those would probably be printed twice which isn't nice.

At the moment, resolving files is a two-step procedure. First, we parse entities but don't apply much analysis. For tables, we already have the exact structure at this point. For views, we don't. In a second step, we resolve all other entities and perform static analysis. A file only enters the second step if all of its imports have finished the first.

Realistically, analyzing views should happen somewhere in the middle between the two steps. Views are tricky to analyze because their columns can depend on other views too, so we have to carefully restrict circular references.

**Comment by LeFrosch** [comment]
Maybe something like this could work: 
```dart
final viewAnalyzedGrouping = [
  ...parseResult.declaredViews,
  ...importedViews,
].groupListsBy((v) => v.parserView != null);

await ViewAnalyzer(
  this,
  availableTables,
  viewAnalyzedGrouping[true] ?? [],
  parseResult.imports,
).resolve(viewAnalyzedGrouping[false] ?? []);
```
It’s probably not the best way to check if a view was analysed before by checking the `parserView` field.

**Comment by simolus3** [comment]
Thinking about some more, I think the easiest approach would be to

- disallow views from reading views declared in another `.drift` file
- resolve views after tables are known.

This solves this problem as the expense of making drift views slightly less expressive, but accessing views from different files has always been a bit flaky and that at least makes it consistent and more useful for reading them in queries.

**Comment by simolus3** [comment]
Fixed in 2a347b2f40702df7f1679c5b9cc03b8d72ada079.
